src/Containers.jsx

```diff
@@ -314,45 +316,58 @@ class Containers extends React.Component {
-            emptyCaption = _("Loading...");
-        else if (this.props.textFilter.length > 0)
-            emptyCaption = _("No containers that match the current filter");
-        else if (this.props.onlyShowRunning)
+        let emptyCaptionPod = _("No containers in this pod");
+        if (this.props.containers === null || this.props.pods === null) {
+            emptyCaption = emptyCaptionPod = _("Loading...");
+        } else if (this.props.textFilter.length > 0) {
+            emptyCaption = emptyCaptionPod = _("No containers that match the current filter");
+        } else if (this.props.onlyShowRunning)
             emptyCaption = _("No running containers");
-
-        const containersStats = this.props.containersStats;
-        let filtered = [];
-        if (this.props.containers !== null)
-            filtered = Object.keys(this.props.containers).filter(id => !this.props.onlyShowRunning || this.props.containers[id].State == "running");
-
-        if (this.props.ownerFilter !== "all") {
-            filtered = filtered.filter(id => {
-                if (this.props.ownerFilter === "system" && !this.props.containers[id].isSystem)
-                    return false;
-                if (this.props.ownerFilter !== "system" && this.props.containers[id].isSystem)
-                    return false;
-                return true;
-            });
+            emptyCaptionPod = _("No running containers in this pod");
         }
 
-        if (this.props.textFilter.length > 0) {
-            const lcf = this.props.textFilter.toLowerCase();
-            filtered = filtered.filter(id => this.props.containers[id].Names[0].toLowerCase().indexOf(lcf) >= 0 ||
-                    this.props.containers[id].Image.toLowerCase().indexOf(lcf) >= 0
-            );
-        }
+        if (this.props.containers !== null && this.props.pods !== null) {
+            if (this.props.containers !== null && this.props.pods !== null)
+                filtered = Object.keys(this.props.containers).filter(id => !this.props.onlyShowRunning || this.props.containers[id].State == "running");
+
+            if (this.props.ownerFilter !== "all") {
+                filtered = filtered.filter(id => {
+                    if (this.props.ownerFilter === "system" && !this.props.containers[id].isSystem)
+                        return false;
+                    if (this.props.ownerFilter !== "system" && this.props.containers[id].isSystem)
+                        return false;
+                    return true;
+                });
+            }
 
-        filtered.sort((a, b) => {
-            // User containers are in front of system ones
-            if (this.props.containers[a].isSystem !== this.props.containers[b].isSystem)
-                return this.props.containers[a].isSystem ? 1 : -1;
-            return this.props.containers[a].Names > this.props.containers[b].Names ? 1 : -1;
-        });
+            if (this.props.textFilter.length > 0) {
```

Comment:
This also does not seem to filter pod names. I think the behaviour we want;
1. If pod name fits the filter, show it with all containers (don't filter those down)
2. If container inside pod fits the filter, show the filter and just containers that fit the name.
3. If nor the pod name nor any container inside the pod fit the filter, hide the whole pod.